Update Gherkin to 4.0.0

koterpillar opened the pull request.

koterpillar committed:
Require new Gherkin parser

koterpillar committed:
Update parsing Gherkin features

koterpillar committed:
Update expected parser errors

danni commented on requirements.txt:
```diff
@@ -1,5 +1,5 @@
 blessings
 future>=0.14.0
-gherkin-official
+gherkin-official>=4
```
Should we pin the major version a bit more tightly to prevent everything from melting?

danni commented on aloe/parser.py:
```diff
@@ -675,20 +675,29 @@ class Feature(HeaderNode, TaggedNode):
+
+        for child in parsed['children']:
+            if child['type'] == 'Background':
+                assert not self.background, "Duplicate background found."
```
Should raise an `AloeSyntaxError` ?

koterpillar replied:
The parser already does that, this would be really unexpected.

koterpillar committed:
Upper bound for Gherkin version

koterpillar committed:
Explain 'not self.background' assertion

Gherkin parser already bails out on seeing multiple backgrounds, so Aloe
doesn't need to do anything meaningful on seeing that.

danni commented:
++
